src/main/scala/mg7/pipeline.scala

```diff
@@ -0,0 +1,209 @@
+package ohnosequences.mg7
+
+import ohnosequences.mg7.loquats._
+import ohnosequences.loquat._
+import ohnosequences.statika._, aws._
+import ohnosequences.datasets._
+import ohnosequences.awstools.s3._
+import com.amazonaws.auth._, profile._
+
+
+trait AnyMG7Pipeline { pipeline =>
+
+  type Parameters <: AnyMG7Parameters
+  val  parameters: Parameters
+
+  val inputSamples: Map[SampleID, S3Resource]
+  val outputS3Folder: (SampleID, StepName) => S3Folder
+
+  val metadata: AnyArtifactMetadata
+  val iamRoleName: String
+  val logsBucketName: String
+
+  /* This trait helps to set these common values */
+  trait CommonConfigDefaults {
+
+    val metadata: AnyArtifactMetadata = pipeline.metadata
+    val iamRoleName: String           = pipeline.iamRoleName
+    val logsBucketName: String        = pipeline.logsBucketName
+  }
+
+  case class  SplitConfig(val size: Int) extends AnySplitConfig  with CommonConfigDefaults
+  case class  BlastConfig(val size: Int) extends AnyBlastConfig  with CommonConfigDefaults
+  case class AssignConfig(val size: Int) extends AnyAssignConfig with CommonConfigDefaults
+  case class  MergeConfig(val size: Int) extends AnyMergeConfig  with CommonConfigDefaults
+  case class  CountConfig(val size: Int) extends AnyCountConfig  with CommonConfigDefaults
+
```

Comment:
These are convenience constructors that take advantage of the pipeline scope and mixin that `CommonConfigDefaults` (because it would be quite inconvenient to define it again and again for each of them)